test/jdk/java/nio/file/Files/CopyInterference.java

```diff
@@ -0,0 +1,151 @@
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/* @test
+ * @bug 8114830
+ * @summary Verify FileAlreadyExistsException is not thrown for REPLACE_EXISTING
+ * @run junit CopyInterference
+ */
+import java.io.InputStream;
+import java.io.IOException;
+import java.nio.file.CopyOption;
+import java.nio.file.FileAlreadyExistsException;
+import java.nio.file.Files;
+import java.nio.file.FileSystemException;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Stream;
+
+import static java.nio.file.StandardCopyOption.*;
+import static java.nio.file.LinkOption.*;
+
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+public class CopyInterference {
+
+    private static final int N_THREADS = 2;
+
+    private static final AtomicBoolean running = new AtomicBoolean(true);
+
+    private static class CopyTask implements Runnable {
+        final Path source;
+        final Path target;
+        final CopyOption[] options;
+
+        CopyTask(Path source, Path target, CopyOption[] options) {
+            this.source = source;
+            this.target = target;
+            this.options = options;
+        }
+
+        @Override
+        public void run() {
+            try {
+                while (running.get()) {
+                    Files.copy(source, target, options);
+                }
+            } catch (FileAlreadyExistsException e) {
+                throw new RuntimeException("Unexpected exception", e);
+            } catch (FileSystemException e) {
+                System.out.printf("Expected FileSystemException: \"%s\"%n",
+                                  e.getMessage());
+            } catch (IOException e) {
+                throw new RuntimeException("Unexpected exception", e);
+            } finally {
+                running.set(false);
+            }
+        }
+    }
+
+    private static Stream<Arguments> pathAndOptionsProvider()
+        throws IOException {
+        Path parent = Path.of(System.getProperty("test.dir", "."));
```

Comment:
I will suggest you to use "java.io.tmpdir" instead of "test.dir" so that all dir/files will get created in 'tmp' directory.

Reply:
> I will suggest you to use "java.io.tmpdir" instead of "test.dir" so that all dir/files will get created in 'tmp' directory.

The problem with java.io.tmpdir is that it will fail up /tmp (or equivalent). Also one advantage of using the test work directory is that it will be archived with the test results so it can, in many tests, help with diagnosing why the test failed.

Reply:
yep it make sense , i suggested because directory/files are not getting cleaned up after test  run.